### PR TITLE
chore(common): Auto-generate minimum-versions.md from min-ver.sh

### LINKED ISSUE #11372: chore(common): make minimum-versions.md
minimum-versions.md (include a link to any app-help section that’d need updating to match)

Ref to #11371.

### PR DESCRIPTION
**Pull request by darcywong00**
Replaces #11386 and fixes #11372

This adds a `min-ver.sh` script that parses `minimum-versions.inc.sh` and `minimum-versions.md.in` to genereate the `minimum-versions.md` file.

I've checked in minimum-versions.md on this PR. Haven't setup any workflows to automate the PRs.

@keymanapp-test-bot skip

### CODE REVIEW

File: resources/build/min-ver.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+function generate_table {
+  echo ""
+  echo "## minimum-versions.inc.sh Variables"
+  echo ""
+  echo "|          KEYMAN Variable          |     Value    |"
+  echo "|-----------------------------------|--------------|"
+
+  # Get all the shell variables named KEYMAN*VERSION
+  strarr=( $(compgen -v KEYMAN | grep VERSION) )
```
Comment (darcywong00) [inline resources/build/min-ver.sh:29]: I found `compgen` can generate a list of shell variables starting with "KEYMAN". Is it available on Windows Git Bash?

Comment (mcdurdin) [reply resources/build/min-ver.sh:29]: yes

image

File: resources/build/min-ver.sh
Comment (mcdurdin) [inline resources/build/min-ver.sh:1]: Can we rename this to `publish-minimum-versions.sh`?

Then we should add a step to at least one of our common test builds to run this script and if any change to minimum-versions.md is detected, then fail the build.  The build fail should include a note on checking that product documentation matches minimum-versions.inc.sh.

**Review by mcdurdin — CHANGES_REQUESTED** [review]
LGTM

We should add a step to our common builds to run min-ver.sh

File: resources/build/min-ver.sh
Comment (darcywong00) [reply resources/build/min-ver.sh:1]: I've renamed to public-minimum-versions.sh.

I'll need some help on the build steps later....

File: resources/build/minimum-versions.md
Comment (mcdurdin) [inline resources/build/minimum-versions.md:1]: Oh, I missed this -- this file should be in /docs (the .in can stay in resources/build)

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.38-alpha](https://keyman.com/downloads/releases/alpha/18.0.38)
